apis/management.cattle.io/v3/catalog_types.go

```diff
@@ -172,3 +172,10 @@ type TemplateContent struct {
 	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#spec-and-status
 	Data string `json:"data,omitempty"`
 }
+
+type MultiClusterApp struct {
+	TemplateName string `json:"multiclusterapp" norman:"required,type=reference[template]"`
+	ProjectID    string `json:"projectID,omitempty" normal:"required,type=reference[project]"`
```

Comment:
no need for cluster id; project id is enough. It should also be a list of project ids as you can deploy the app to many of them

Reply:
yes, I had created a very rough WIP PR yesterday. I'll go over the design docs and discuss to get a more clear idea :)